### ISSUE TITLE
Template not displaying (shows default nginx 503 page)

### ISSUE THREAD

**Issue by kyleschmolze** [dokku/dokku-maintenance#16]
## Description of problem
Hi there! I'm on fully upgraded Dokku (0.28.1), and just installed dokku-maintenance fresh. When I enable maintenance mode, my app serves this charming template:

![image](https://user-images.githubusercontent.com/1200038/191395690-59bd9856-65a5-4d6e-9f29-9a43f9d5912a.png)

I see that in `/templates`, there's a default `maintenance.html` which (I'm guessing) is supposed to be served. I also tried setting a custom page, and even though I received this helpful message:
```
----> Importing custom maintenance page...
maintenance.html
       Done
```
I still only get that ugly white nginx page :/ 

### How reproducible
I'm not 100% sure, but my guess is that there's a bug with 0.28.1 of Dokku?

### How (deb/make/rpm) and where (AWS, VirtualBox, physical, etc.) was Dokku installed?: 
Just upgraded to 0.28 today using apt-get. 

### Additional information
Let me know what else would be helpful!

Thank you so much for the great plugin, I'm excited to get it working :) 

### `dokku report jumpoffcampus` output
```
-----> uname: Linux jumpoffcampus 5.4.0-52-generic #57-Ubuntu SMP Thu Oct 15 10:57:00 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
-----> memory: 
                     total        used        free      shared  buff/cache   available
       Mem:           3936        1882         765         153        1288        1626
       Swap:             0           0           0
-----> docker version: 
       Client: Docker Engine - Community
        Version:           19.03.13
        API version:       1.40
        Go version:        go1.13.15
        Git commit:        4484c46d9d
        Built:             Wed Sep 16 17:02:52 2020
        OS/Arch:           linux/amd64
        Experimental:      false
       
       Server: Docker Engine - Community
        Engine:
         Version:          19.03.13
         API version:      1.40 (minimum version 1.12)
         Go version:       go1.13.15
         Git commit:       4484c46d9d
         Built:            Wed Sep 16 17:01:20 2020
         OS/Arch:          linux/amd64
         Experimental:     false
        containerd:
         Version:          1.4.12
         GitCommit:        7b11cfaabd73bb80907dd23182b9347b4245eb5d
        runc:
         Version:          1.0.2
         GitCommit:        v1.0.2-0-g52b36a2
        docker-init:
         Version:          0.18.0
         GitCommit:        fec3683
-----> docker daemon info: 
       Client:
        Debug Mode: true
       
       Server:
        Containers: 13
         Running: 10
         Paused: 0
         Stopped: 3
        Images: 75
        Server Version: 19.03.13
        Storage Driver: overlay2
         Backing Filesystem: extfs
         Supports d_type: true
         Native Overlay Diff: true
        Logging Driver: json-file
        Cgroup Driver: cgroupfs
        Plugins:
         Volume: local
         Network: bridge host ipvlan macvlan null overlay
         Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
        Swarm: inactive
        Runtimes: runc
        Default Runtime: runc
        Init Binary: docker-init
        containerd version: 7b11cfaabd73bb80907dd23182b9347b4245eb5d
        runc version: v1.0.2-0-g52b36a2
        init version: fec3683
        Security Options:
         apparmor
         seccomp
          Profile: default
        Kernel Version: 5.4.0-52-generic
        Operating System: Ubuntu 20.04.3 LTS
        OSType: linux
        Architecture: x86_64
        CPUs: 2
        Total Memory: 3.844GiB
        Name: jumpoffcampus
        ID: PEEB:CHZO:EMP7:XJR5:2ZB7:QM5F:TPDW:PHRP:MO73:X7YR:V5PN:LJDA
        Docker Root Dir: /var/lib/docker
        Debug Mode: false
        Registry: https://index.docker.io/v1/
        Labels:
        Experimental: false
        Insecure Registries:
         127.0.0.0/8
        Live Restore Enabled: false
       
-----> git version: git version 2.25.1
-----> sigil version: 0.9.0build+bc921b7
-----> herokuish version: 
       herokuish: 0.5.37
       buildpacks:
         heroku-buildpack-multi     v1.2.0
         heroku-buildpack-ruby      v244
         heroku-buildpack-nodejs    v198
         heroku-buildpack-clojure   v90
         heroku-buildpack-python    v214
         heroku-buildpack-java      v72
         heroku-buildpack-gradle    v38
         heroku-buildpack-scala     v94
         heroku-buildpack-play      v26
         heroku-buildpack-php       v223
         heroku-buildpack-go        v166
         heroku-buildpack-nginx     v16
         buildpack-null             v3
-----> dokku version: dokku version 0.28.1
-----> plugn version: plugn: 0.12.0build+3a27594
-----> dokku plugins: 
         00_dokku-standard    0.28.1 enabled    dokku core standard plugin
         20_events            0.28.1 enabled    dokku core events logging plugin
         app-json             0.28.1 enabled    dokku core app-json plugin
         apps                 0.28.1 enabled    dokku core apps plugin
         builder              0.28.1 enabled    dokku core builder plugin
         builder-dockerfile   0.28.1 enabled    dokku core builder-dockerfile plugin
         builder-herokuish    0.28.1 enabled    dokku core builder-herokuish plugin
         builder-lambda       0.27.0 enabled    dokku core builder-lambda plugin
         builder-null         0.28.1 enabled    dokku core builder-null plugin
         builder-pack         0.28.1 enabled    dokku core builder-pack plugin
         buildpacks           0.28.1 enabled    dokku core buildpacks plugin
         caddy-vhosts         0.28.1 enabled    dokku core caddy-vhosts plugin
         certs                0.28.1 enabled    dokku core certificate management plugin
         checks               0.28.1 enabled    dokku core checks plugin
         common               0.28.1 enabled    dokku core common plugin
         config               0.28.1 enabled    dokku core config plugin
         cron                 0.28.1 enabled    dokku core cron plugin
         docker-options       0.28.1 enabled    dokku core docker-options plugin
         domains              0.28.1 enabled    dokku core domains plugin
         enter                0.28.1 enabled    dokku core enter plugin
         git                  0.28.1 enabled    dokku core git plugin
         letsencrypt          0.12.1 enabled    Automated installation of let's encrypt TLS certificates
         logs                 0.28.1 enabled    dokku core logs plugin
         maintenance          0.7.2 enabled    Maintenance mode for apps
         network              0.28.1 enabled    dokku core network plugin
         nginx-vhosts         0.28.1 enabled    dokku core nginx-vhosts plugin
         plugin               0.28.1 enabled    dokku core plugin plugin
         postgres             1.19.2 enabled    dokku postgres service plugin
         proxy                0.28.1 enabled    dokku core proxy plugin
         ps                   0.28.1 enabled    dokku core ps plugin
         redis                1.19.1 enabled    dokku redis service plugin
         registry             0.28.1 enabled    dokku core registry plugin
         repo                 0.28.1 enabled    dokku core repo plugin
         resource             0.28.1 enabled    dokku core resource plugin
         run                  0.28.1 enabled    dokku core run plugin
         scheduler            0.28.1 enabled    dokku core scheduler plugin
         scheduler-docker-local 0.28.1 enabled    dokku core scheduler-docker-local plugin
         scheduler-null       0.28.1 enabled    dokku core scheduler-null plugin
         shell                0.28.1 enabled    dokku core shell plugin
         ssh-keys             0.28.1 enabled    dokku core ssh-keys plugin
         storage              0.28.1 enabled    dokku core storage plugin
         trace                0.28.1 enabled    dokku core trace plugin
         traefik-vhosts       0.28.1 enabled    dokku core traefik-vhosts plugin
=====> jumpoffcampus app-json information
       App json computed selected:    app.json
       App json global selected:      app.json
       App json selected:             
=====> jumpoffcampus app information
       App created at:                1663724518
       App deploy source:             
       App deploy source metadata:    
       App dir:                       /home/dokku/jumpoffcampus
       App locked:                    true
=====> jumpoffcampus builder information
       Builder build dir:             
       Builder computed build dir:    
       Builder computed selected:     
       Builder global build dir:      
       Builder global selected:       
       Builder selected:              
=====> jumpoffcampus builder-dockerfile information
       Builder dockerfile computed dockerfile path: Dockerfile               
       Builder dockerfile global dockerfile path: Dockerfile               
       Builder dockerfile dockerfile path:                          
=====> jumpoffcampus builder-lambda information
       Builder lambda computed lambdayml path: lambda.yml               
       Builder lambda global lambdayml path: lambda.yml               
       Builder lambda lambdayml path:                          
=====> jumpoffcampus builder-pack information
       Builder pack computed projecttoml path: project.toml             
       Builder pack global projecttoml path: project.toml             
       Builder pack projecttoml path:                          
=====> jumpoffcampus buildpacks information
       Buildpacks computed stack:     gliderlabs/herokuish:latest-20
       Buildpacks global stack:       
       Buildpacks list:               
       Buildpacks stack:              
=====> jumpoffcampus ssl information
       Ssl dir:                       /home/dokku/jumpoffcampus/tls
       Ssl enabled:                   true                     
       Ssl hostnames:                 *.jocamp.us *.jumpoffcampus.com jocamp.us jumpoffcampus.com
       Ssl expires at:                Oct 19 21:22:51 2022 GMT 
       Ssl issuer:                    C = US, O = Let's Encrypt, CN = R3
       Ssl starts at:                 Jul 21 21:22:52 2022 GMT 
       Ssl subject:                   subject=CN = *.jumpoffcampus.com
       Ssl verified:                  self signed              
=====> jumpoffcampus checks information
       Checks disabled list:          none                     
       Checks skipped list:           none                     
=====> jumpoffcampus cron information
       Cron task count:               3
=====> jumpoffcampus docker options information
       Docker options build:          --link dokku.postgres.jumpoffcampus-db:dokku-postgres-jumpoffcampus-db --link dokku.redis.jumpoffcampus-redis:dokku-redis-jumpoffcampus-redis 
       Docker options deploy:         --link dokku.postgres.jumpoffcampus-db:dokku-postgres-jumpoffcampus-db --link dokku.redis.jumpoffcampus-redis:dokku-redis-jumpoffcampus-redis --restart=on-failure:10 
       Docker options run:            --link dokku.postgres.jumpoffcampus-db:dokku-postgres-jumpoffcampus-db --link dokku.redis.jumpoffcampus-redis:dokku-redis-jumpoffcampus-redis 
=====> jumpoffcampus domains information
       Domains app enabled:           true                     
       Domains app vhosts:            jocamp.us *.jocamp.us *.jocamp.us jumpoffcampus.com *.jumpoffcampus.com
       Domains global enabled:        true                     
       Domains global vhosts:         159.223.149.60           
=====> jumpoffcampus git information
       Git deploy branch:             master                   
       Git global deploy branch:      master                   
       Git keep git dir:              false                    
       Git rev env var:               GIT_REV                  
       Git sha:                       1782a3b                  
       Git last updated at:           1663721176               
=====> jumpoffcampus logs information
       Logs computed max size:        10m
       Logs global max size:          10m
       Logs global vector sink:       
       Logs max size:                 
       Logs vector sink:              
=====> jumpoffcampus maintenance information
       Maintenance enabled:           true                     
=====> jumpoffcampus network information
       Network attach post create:           
       Network attach post deploy:           
       Network bind all interfaces:          false
       Network computed attach post create:  
       Network computed attach post deploy:  
       Network computed bind all interfaces: false
       Network computed initial network:     
       Network computed tld:                 
       Network global attach post create:    
       Network global attach post deploy:    
       Network global bind all interfaces:   false
       Network global initial network:       
       Network global tld:                   
       Network initial network:              
       Network static web listener:          
       Network tld:                          
       Network web listeners:                172.17.0.12:5000
=====> jumpoffcampus nginx information
       Nginx access log format:                                
       Nginx access log path:         /var/log/nginx/jumpoffcampus-access.log
       Nginx bind address ipv4:                                
       Nginx bind address ipv6:       ::                       
       Nginx client max body size:                             
       Nginx disable custom config:   false                    
       Nginx error log path:          /var/log/nginx/jumpoffcampus-error.log
       Nginx global hsts:             true                     
       Nginx computed hsts:           true                     
       Nginx hsts:                                             
       Nginx hsts include subdomains: true                     
       Nginx hsts max age:            15724800                 
       Nginx hsts preload:            false                    
       Nginx proxy buffer size:       4096                     
       Nginx proxy buffering:         on                       
       Nginx proxy buffers:           8 4096                   
       Nginx proxy busy buffers size: 8192                     
       Nginx proxy read timeout:      60s                      
       Nginx last visited at:         1663724709               
       Nginx x forwarded for value:   $remote_addr             
       Nginx x forwarded port value:  $server_port             
       Nginx x forwarded proto value: $scheme                  
       Nginx x forwarded ssl:                                  
=====> jumpoffcampus proxy information
       Proxy enabled:                 true
       Proxy port map:                http:80:5000 https:443:5000
       Proxy type:                    nginx
=====> jumpoffcampus ps information
       Deployed:                      true
       Processes:                     1
       Ps can scale:                  true
       Ps computed procfile path:     Procfile
       Ps global procfile path:       Procfile
       Ps procfile path:              
       Ps restart policy:             on-failure:10
       Restore:                       true
       Running:                       true
       Status web 1:                  running (CID: 57db1b16254)
=====> jumpoffcampus registry information
       Registry computed image repo:      dokku/jumpoffcampus
       Registry computed push on release: false
       Registry computed server:          
       Registry global push on release:   
       Registry global server:            
       Registry image repo:               
       Registry push on release:          
       Registry server:                   
       Registry tag version:              
=====> jumpoffcampus resource information
=====> jumpoffcampus scheduler information
       Scheduler computed selected:   docker-local
       Scheduler global selected:     docker-local
       Scheduler selected:            
=====> jumpoffcampus scheduler-docker-local information
       Scheduler docker local disable chown:                          
       Scheduler docker local init process: true                     
       Scheduler docker local parallel schedule count:                          
=====> jumpoffcampus storage information
       Storage build mounts:                                   
       Storage deploy mounts:                                  
       Storage run mounts:   
```

### Output of dokku nginx:show-config jumpoffcampus
```
server {
  listen      [::]:80;
  listen      80;
  server_name jocamp.us *.jocamp.us *.jocamp.us jumpoffcampus.com *.jumpoffcampus.com; 
  access_log  /var/log/nginx/jumpoffcampus-access.log;
  error_log   /var/log/nginx/jumpoffcampus-error.log;

  include /home/dokku/jumpoffcampus/nginx.conf.d/*.conf;
  location / {
    return 301 https://$host:443$request_uri;
  }

}

server {
  listen      [::]:443 ssl http2;
  listen      443 ssl http2;
  
  server_name jocamp.us *.jocamp.us *.jocamp.us jumpoffcampus.com *.jumpoffcampus.com; 
  access_log  /var/log/nginx/jumpoffcampus-access.log;
  error_log   /var/log/nginx/jumpoffcampus-error.log;

  ssl_certificate           /home/dokku/jumpoffcampus/tls/server.crt;
  ssl_certificate_key       /home/dokku/jumpoffcampus/tls/server.key;
  ssl_protocols             TLSv1.2 TLSv1.3;
  ssl_prefer_server_ciphers off;

  keepalive_timeout   70;
  

  location    / {

    gzip on;
    gzip_min_length  1100;
    gzip_buffers  4 32k;
    gzip_types    text/css text/javascript text/xml text/plain text/x-component application/javascript application/x-javascript application/json application/xml  application/rss+xml font/truetype application/x-font-ttf font/opentype application/vnd.ms-fontobject image/svg+xml;
    gzip_vary on;
    gzip_comp_level  6;

    proxy_pass  http://jumpoffcampus-5000;
    http2_push_preload on; 
    proxy_http_version 1.1;
    proxy_read_timeout 60s;
    proxy_buffer_size 4096;
    proxy_buffering on;
    proxy_buffers 8 4096;
    proxy_busy_buffers_size 8192;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $http_connection;
    proxy_set_header Host $http_host;
    proxy_set_header X-Forwarded-For $remote_addr;
    proxy_set_header X-Forwarded-Port $server_port;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header X-Request-Start $msec;
    
  }

  

  error_page 400 401 402 403 405 406 407 408 409 410 411 412 413 414 415 416 417 418 420 422 423 424 426 428 429 431 444 449 450 451 /400-error.html;
  location /400-error.html {
    root /var/lib/dokku/data/nginx-vhosts/dokku-errors;
    internal;
  }

  error_page 404 /404-error.html;
  location /404-error.html {
    root /var/lib/dokku/data/nginx-vhosts/dokku-errors;
    internal;
  }

  error_page 500 501 503 504 505 506 507 508 509 510 511 /500-error.html;
  location /500-error.html {
    root /var/lib/dokku/data/nginx-vhosts/dokku-errors;
    internal;
  }

  error_page 502 /502-error.html;
  location /502-error.html {
    root /var/lib/dokku/data/nginx-vhosts/dokku-errors;
    internal;
  }
  include /home/dokku/jumpoffcampus/nginx.conf.d/*.conf;
}

upstream jumpoffcampus-5000 {

  server 172.17.0.12:5000;
}
```

**Comment by robswc** [comment]
Just wanted to say I'm experiencing the same issue.
May do a deep dive to see if there's any quick fix.

**Comment by SteelAlloy** [comment]
Same bug here.
It only occurs with https, but when using http the maintenance page is correctly displayed.

**Comment by mrbrdo** [comment]
I have the same issue, did anyone manage to fix this? For me it's on **http as well**, not just https. Don't even have https configured yet.

**Comment by SteelAlloy** [comment]
> Same bug here. It only occurs with https, but when using http the maintenance page is correctly displayed.

Turns out it doesn't work when I disable letsencrypt.

However, when https is enabled through letencrypt, `curl http://mypage.com` returns the maintenance page while `curl https://mypage.com` gives the 503 error page.

**Comment by clreinki** [comment]
Just adding myself as another user affected by this.  Currently running v0.28.4.  I am NOT using the letsencrypt plugin, though I do have an enterprise SSL cert.

I was going to attach my nginx:show-config output, however I reviewed it line by line and it is identical to the OP's config (but with different hostnames).

**Comment by josegonzalez** [comment]
Is everyone here on Ubuntu 22.04 or higher?

**Comment by clreinki** [comment]
> Is everyone here on Ubuntu 22.04 or higher?

I am on Ubuntu 20.04 LTS (though planning to upgrade soon)

**Comment by davidrunger** [comment]
I haven't chimed in yet, but I'm following this issue because I am also affected. I am on Ubuntu 22.04.1.

**Comment by davidcelis** [comment]
I'm experiencing the same issue; just installed this plugin fresh and can't get my custom maintenance page served. I'm also on Ubuntu 22.04.1 and, while I'm not using LetsEncrypt, I am using a custom origin certificate from Cloudflare and enforcing SSL on my site.

**Comment by SteelAlloy** [comment]
Quick and (dirty) easy fix :

Remove `503` from nginx.conf
```
  error_page 500 501 504 505 506 507 508 509 510 511 /500-error.html;
  location /500-error.html {
    root /var/lib/dokku/data/nginx-vhosts/dokku-errors;
    internal;
  }
```

**Comment by josegonzalez** [comment]
Closed by #17.
